src/components/query/QueryDirective.js

```diff
@@ -352,8 +353,16 @@
 
     // Watcher/listener
     $scope.layers = $scope.map.getLayers().getArray();
-    $scope.layerFilter = gaLayerFilters.selectByRectangle;
-    $scope.$watchCollection('layers | filter:layerFilter', function(layers) {
+    $scope.selectByRectangleFilter = gaLayerFilters.selectByRectangle;
+    $scope.$watchCollection('layers | filter:selectByRectangleFilter',
```

Comment:
Yes. We now have 2 sets of layers. 1 for query tool (searchable layers) and another for the select by rectangle tool (queryable layers)